tests/unit/session_test.py

```diff
@@ -18,4 +24,49 @@ def test_session_connection_refused(free_port_num):
     error_msg = '^ssh connect failed: Connection refused$'
     ssh_session = Session()
     with pytest.raises(LibsshSessionException, match=error_msg):
-        ssh_session.connect(host='127.0.0.1', port=free_port_num)
+        ssh_session.connect(host=LOCALHOST, port=free_port_num)
+
+
+def test_session_log_level_debug(caplog, free_port_num):
+    """Test setting the log level to DEBUG should reveal copyright information."""
+    ssh_session = Session()
+    ssh_session.set_log_level(logging.DEBUG)
+
+    # the connection will fail but first log lands before that
+    with pytest.raises(LibsshSessionException):
+        ssh_session.connect(host=LOCALHOST, port=free_port_num)
+
+    found_copyright = False
+    for record in caplog.records:
+        # This log message is shown at different log levels in different libssh versions
+        if record.levelname in {'DEBUG', 'INFO'} and 'and libssh contributors.' in record.msg:
+            found_copyright = True
+    assert found_copyright
```

Comment:
Let's make the check more straightforward here:
```suggestion
    expected_copyright_substring = 'and libssh contributors.'
    # This log message is shown at different log levels
    # in different libssh versions:
    expected_copyright_log_levels = {'DEBUG', 'INFO'}

    informational_log_messages = (
        record.msg
        for record in caplog.records
        if record.levelname in expected_copyright_log_levels
    )

    assert expected_copyright_substring in informational_log_messages
```

Reply:
Adjusted! Thanks!